compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaUnionTypeSymbol.java

```diff
@@ -147,6 +154,32 @@ public String signature() {
         return this.signature;
     }
 
+    @Override
+    public boolean isEnum() {
+        if (this.enumSymbol != null) {
+            return true;
+        }
+
+        return this.getBType().tsymbol.getKind() == SymbolKind.ENUM;
+    }
+
+    @Override
+    public Optional<EnumSymbol> getEnumSymbol() {
```

Comment:
I'm not sure this is ideal. isEnum kinda makes sense, but why do we need to allow retrieving the enum symbol like this? In a way, it's like having a reference to the definition from the type symbol.

Reply:
Yes. It's kinda exposing the information related to the enum symbol. Currently we may not have such things, but  having this may allow us to expose those infromation right?

Reply:
and this is not a way of referencing. For referencing we have a structure as TypeRefType

Reply:
I meant having `enumSymbol` in `BallerinaUnionTypeSymbol`. This isn't quite right, IMO. The type doesn't need to how it was defined.



> Yes. It's kinda exposing the information related to the enum symbol. Currently, we may not have such things, but having this may allow us to expose that information right?

I don't think that's the correct way. If someone wants enum symbol they can still get it from the model, right? I don't think the type symbol should expose it.

Reply:
>I meant having enumSymbol in BallerinaUnionTypeSymbol. This isn't quite right, IMO. The type doesn't need to how it was defined.
yeah. What if we have introduced some attributes only related to enums and the ballerina extension developer needs to get them? We can't expose that only-enum information via the BallerinaUnionTypeSymbol right?

> I don't think that's the correct way. If someone wants enum symbol they can still get it from the model, right? I don't think the type symbol should expose it.

How can they get an enum symbol without the location?

Reply:
> > I meant having enumSymbol in BallerinaUnionTypeSymbol. This isn't quite right, IMO. The type doesn't need to how it was defined.
> 
> yeah. What if we have introduced some attributes only related to enums and the ballerina extension developer needs to get them? We can't expose that only-enum information via the BallerinaUnionTypeSymbol right?

Can you give an example for this?

> 
> > I don't think that's the correct way. If someone wants enum symbol they can still get it from the model, right? I don't think the type symbol should expose it.
> 
> How can they get an enum symbol without the location?

Can't they filter out from module symbols by name?

Reply:
> We can't expose that only-enum information via the BallerinaUnionTypeSymbol right?

Yes, we can't. I can't think of such info anyway. We shouldn't mix up enums and types. An enum is just another way to define a union of string singletons. From a type symbol point of view, they are basically the same.

Reply:
> Can you give an example for this?

what about annotation attachments? We shouldn't link them with the UnionTypeSymbol right?

> Can't they filter out module symbols by name?

Yeah, we can, but we need the name to filter out right? What if we need to get the enum that is in another module, and we have referenced an enum value (e.x: OPEN) and we need to know the other sibling values of the enum (which is referenced in new type?

```ballerina
//  defined in another module `test0.module1`'
public enum State {
    OPEN,
    CLOSED
}

// main.bal
import test0.module1;

type ConnectionState State;

function test() {
    string x = module1.OPEN;
}
```
Let's say there is a ballerina extension writer who needs to get only the sibling values of `OPEN`.
 
Also, isn't it costly to use the filter when there are so many symbols?

> Yes, we can't. I can't think of such info anyway. We shouldn't mix up enums and types. An enum is just another way to define a union of string singletons. From a type symbol point of view, they are basically the same.

Yes. This is true, but here we are just showing the relationship with the associated enum defined somewhere

Reply:
> what about annotation attachments? We shouldn't link them with the UnionTypeSymbol right?

```ballerina
annotation annot on type;

const RED = "RED";
const BLUE = "BLUE";

@annot
type Colour RED|BLUE;
```

is the same as

```ballerina
annotation annot on type;

@annot
enum Colour {
    RED,
    BLUE
}
```

so we shouldn't have to special case anything for enums.



> Yeah, we can, but we need the name to filter out right? 

If you have the union type you have the name?

> What if we need to get the enum that is in another module, and we have referenced an enum value (e.x: OPEN) and we need to know the other sibling values of the enum (which is referenced in new type?

How does adding the enum symbol to the union type help with this? The union type corresponds to the enum, not the member?

> Also, isn't it costly to use the filter when there are so many symbols?

It's just filtering based on a string? Anyway, the API being well-defined is more important IMO.